spring-integration-event/src/main/java/org/springframework/integration/event/inbound/ApplicationEventListeningMessageProducer.java

```diff
@@ -116,20 +129,45 @@ public void onApplicationEvent(ApplicationEvent event) {
 		}
 	}
 
+	@Override
+	protected Object evaluatePayloadExpression(Object rootObject) {
+		if (!this.payloadExpressionSet && rootObject instanceof PayloadApplicationEvent) {
+			return ((PayloadApplicationEvent) rootObject).getPayload();
```

Comment:
I am not convinced this is valid.

The `PayloadApplicationEvent` is `Mainly intended for internal use within the framework.`.

In any case, I don't think the `Event` adapter should unwrap the contents; people will (reasonably IMHO) expect the event adapter to emit messages that have `ApplicationEvent` payloads.

For the same reason, I am not convinced it should be possible to apply an expression to the event.